tests/py/common/node/node.py

```diff
@@ -0,0 +1,90 @@
+import atexit
+import json
+import shutil
+from dataclasses import dataclass
+from pathlib import Path
+from subprocess import Popen
+from typing import Optional
+
+import web3
+from eth_account.signers.local import LocalAccount
+from web3 import Web3
+from web3.eth import Eth
+from web3.net import Net
+
+from common.util.predicates import YES
+from common.util.wait import wait
+
+_localhost = '127.0.0.1'
+
+
+class Node:
+    class InitMode:
+        pass
+
+    @dataclass
+    class ManagedProcessInitMode(InitMode):
+        executable_path: str
+        clean_data = True
+
+    @dataclass
+    class RemoteInitMode(InitMode):
+        host: str = _localhost
+
+    def __init__(self, cfg_file_path, wallet_file_path, mode: InitMode, default_w3_provider_type='http'):
+        wallet_file_path, cfg_file_path = Path(wallet_file_path), Path(cfg_file_path)
+        with open(cfg_file_path, mode="r") as f:
+            cfg = json.load(f)
+        self._w3_by_type = {}
+        self._proc = None
+
+        with open(wallet_file_path, mode="r") as f:
+            self.account: LocalAccount = web3.Account.from_key(json.load(f)["node_secret"])
+        self.w3: Optional[Web3] = None
+        self.eth: Optional[Eth] = None
+        self.net: Optional[Net] = None
+
+        atexit.register(self.destructor)
+
+        net_host = _localhost
+        if isinstance(mode, Node.ManagedProcessInitMode):
+            if mode.clean_data:
+                shutil.rmtree(Path(cfg["db_path"]), ignore_errors=True)
```

Comment:
there is no more `db_path` I changed it to`data_path`. Please check https://github.com/Taraxa-project/taraxa-node/pull/940